scripts/runSchoolApi.js

```diff
@@ -0,0 +1,116 @@
+'use strict'
+
+const axios = require('axios')
+
+const path = '/admin/api/v1'
+const endPointSchools = '/admin/schools'
+const endPointUsers = '/admin/users'
+const federalStateNames = {
+	nbc: 'Niedersachsen',
+	brb: 'Brandenburg',
+}
+const users = {
+	admin1_nbc: 'administrator',
+	teacher1_nbc: 'teacher',
+	student1_nbc: 'student',
+}
+
+const headers = {
+	'Content-Type': 'application/json',
+}
+
+const extractSubdomain = url => {
+	try {
+		const parsedUrl = new URL(url)
+		const subdomain = parsedUrl.hostname.split('.')[1]
+		return subdomain
+	} catch (error) {
+		console.error('Error parsing URL:', error.message)
+		return null
+	}
+}
+
+const createSchool = async (schoolUrl, headers) => {
+	try {
+		const payload = {
+			name: `cypress-automated-tests`,
+			federalStateName: federalStateNames[extractSubdomain(schoolUrl)],
+		}
+
+		console.log(payload)
+
+		const response = await axios.post(schoolUrl, payload, { headers })
+		const { id } = response.data
+		console.log('School created with ID:', id)
+		return { id }
+	} catch (error) {
+		console.error('Error creating school:', error.message)
+		throw error
+	}
+}
+
+const generateRandomUserEmail = () => {
+	const commonNames = ['john', 'emma', 'michael', 'sarah', 'david']
+	const randomName = commonNames[Math.floor(Math.random() * commonNames.length)]
+	const randomNumber = Math.floor(Math.random() * 1000)
+	const emailDomain = 'cypress-mail.de'
+	return `${randomName}${randomNumber}@${emailDomain}`
+}
+
+const createUsers = async (baseUrl, apiKeys, schoolId, userType) => {
+	try {
+		const schoolUrl = `${baseUrl
+			.toLowerCase()
+			.replace(/\/$/, '')}${path}${endPointSchools}`
+		const userUrl = `${baseUrl
+			.toLowerCase()
+			.replace(/\/$/, '')}${path}${endPointUsers}`
```

Comment:
I would extract this logic in a separate method getEndpointUrl

Reply:
Adapted the changes